tiledb/sm/filesystem/azure.cc

```diff
@@ -42,6 +42,13 @@
 
 using namespace tiledb::common;
 
+#if _WIN32 && defined(min)
```

Comment:
Is this necessary? The SDK sets `/DNOMINMAX` for windows, and we use `std::min` in this file.

Reply:
Seems to be a macro active from somewhere.
That, or something else, is necessary.
From azure.cc, without it active,
```
  *nbytes_filled =
      std::min(write_cache_max_size_ - write_cache_buffer->size(), length);
```
becomes
```
  *nbytes_filled =
      std::(((write_cache_max_size_ - write_cache_buffer->size()) < (length)) ? (write_cache_max_size_ - 
   write_cache_buffer->size()) : (length));
```

Reply:
Adding the template-type to `min` or `max` might be enough to stop collisions with the Windows macros. E.g.:
`*nbytes_filled = std::min<uint64_t>(...)`.

Reply:
Pretty sure the preprocessor will preprocess the min/max before the c++ parser ever sees the code.

Reply:
I see, so `Windows.h` is leaking in from the SDK? The undef seems reasonably safe because it is only in this .cc file, but please add a brief comment about why. Marking resolved.

Reply:
According this post: https://stackoverflow.com/questions/11544073/how-do-i-deal-with-the-max-macro-in-windows-h-colliding-with-max-in-std

We should be able to `#define NOMINMAX` to avoid this, but I don't know enough about windows build flags to know if the `/DNOMINMAX` is already attempting to do this.

Reply:
Sorry, didn't see the further discussion -- `#define NOMINMAX` localized here would be fine too, I think.

Reply:
@d-hoke One more clarification, sorry: If I'm looking at the right macro, it is defined as `#define min(a, b) ...`, so the template type would be enough for the preprocessor to fail the substitution. Either way works.

Reply:
Changed to #define NOMINMAX in azure.cc before any other includes, subject to defined(_WIN32).